core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -263,29 +264,30 @@ sealed abstract class Doc extends Product with Serializable {
    * the widest possible variant. This is the same
    * as render(Int.MaxValue) except it is more efficient.
    */
-  def renderWideStream: Stream[String] = {
+  private def renderWideOptionStream: Stream[Option[String]] = {
     @tailrec
-    def loop(pos: Int, lst: List[(Int, Doc)]): Stream[String] = lst match {
+    def loop(pos: Int, lst: List[(Int, Doc)]): Stream[Option[String]] = lst match {
       case Nil => Stream.empty
       case (i, Empty) :: z => loop(pos, z)
+      case (_, Fail) :: _ => None #:: Stream.empty
```

Comment:
if we didn't have the Fail node, we wouldn't need to change this.... I wonder if we could avoid it...

Reply:
Done.